src/v3/src/components/LaunchAuthenticatorButton/LaunchAuthenticatorButton.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { Box, Button as ButtonMui, useTheme } from '@mui/material';
+import { h } from 'preact';
+
+import { useOnSubmit } from '../../hooks';
+import {
+  ClickHandler,
+  LaunchAuthenticatorButtonElement,
+  UISchemaElementComponent,
+} from '../../types';
+import { getTranslation } from '../../util';
+import { OktaVerifyIcon } from '../Icon';
+
+const LaunchAuthenticatorButton: UISchemaElementComponent<{
+  uischema: LaunchAuthenticatorButtonElement
+}> = ({ uischema }) => {
+  const theme = useTheme();
+
+  const onSubmitHandler = useOnSubmit();
+  const {
+    translations = [],
+    options: {
+      step,
+    },
+  } = uischema;
+
+  const handleClick: ClickHandler = async () => {
+    onSubmitHandler({
+      step,
+    });
+  };
+
+  const label = getTranslation(translations, 'label');
+  const iconDescription = getTranslation(translations, 'icon-description') || '';
+
+  return (
+    <ButtonMui
```

Comment:
Do you need to use the `loading` flag here so when the request is pending, the spinner shows on the button? 

If so, will also need to add the logic to the button to display the spinner (or at least disable the button) when loading = true.

Reply:
There isn't a spinner in the current gen2 widget, however if we do want a spinner for this newer widget we might need design input that we can implement in a separate PR. The reason is that we need to work out how to display the spinner in the loading state (i.e. do we show the spinner and the icon, hide the icon when spinner is there, or something else?)

Reply:
Ohh yes, that makes total sense, does v2 disable the button after clicking? Or is there no harm in double clicks?

Reply:
v2 does not disable the button after clicking, although we may actually want to do that. I don't think there's a ton of harm in double clicking however.